themes/bootstrap3/js/searchbox_controls.js

```diff
@@ -155,31 +164,150 @@ VuFind.register('searchbox_controls', function SearchboxControls() {
+          // Global wildcard fallback:
+          if (typeof(formattingRules["*"]) !== "undefined") {
+            return formattingRules["*"];
+          }
+        }
+        return "none";
+      };
+      const typeahead = new Autocomplete({
+        rtl: $(document.body).hasClass("rtl"),
+        maxResults: 10,
+        loadingString: VuFind.translate('loading_ellipsis'),
+      });
+
+      let cache = {};
+      const input = $searchbox[0];
+      typeahead(input, function vufindACHandler(query, callback) {
+        const classParams = extractClassParams(input);
+        const searcher = classParams.searcher;
+        const selectedType = classParams.type
+          ? classParams.type
+          : $(typeFieldSelector ? typeFieldSelector : '#searchForm_type').val();
+        const type = (typePrefix ? typePrefix : "") + selectedType;
+        const formattingRule = getFormattingRule(type);
+
+        const cacheKey = searcher + "|" + type;
+        if (typeof cache[cacheKey] === "undefined") {
+          cache[cacheKey] = {};
+        }
+
+        if (typeof cache[cacheKey][query] !== "undefined") {
+          callback(cache[cacheKey][query]);
+          return;
+        }
+
+        var hiddenFilters = [];
+        $('#searchForm').find('input[name="hiddenFilters[]"]').each(function hiddenFiltersEach() {
+          hiddenFilters.push($(this).val());
+        });
+
+        $.ajax({
+          url: VuFind.path + '/AJAX/JSON',
+          data: {
+            q: query,
+            method: 'getACSuggestions',
+            searcher: searcher,
+            type: type,
+            hiddenFilters,
+          },
+          dataType: 'json',
+          success: function autocompleteJSON(json) {
+            const highlighted = json.data.suggestions.map(
+              (item) => ({
+                text: item.replaceAll("&", "&amp;")
+                  .replaceAll("<", "&lt;")
+                  .replaceAll(">", "&gt;")
+                  .replaceAll(query, `<b>${query}</b>`),
+                value: formattingRule === 'phrase'
+                  ? '"' + item.replaceAll('"', '\\"') + '"'
+                  : item,
+              })
+            );
+            cache[cacheKey][query] = highlighted;
+            callback(highlighted);
+          }
+        });
+      });
+
+      // Bind autocomplete auto submit
+      if ($searchbox.hasClass("ac-auto-submit")) {
+        input.addEventListener("ac-select", (event) => {
+          const value = typeof event.detail === "string"
+            ? event.detail
+            : event.detail.value;
+          input.value = value;
+          input.form.submit();
+        });
+      }
+    });
+  }
+
+  function setupSearchResetButton() {
+    _resetButton = document.getElementById("searchForm-reset");
 
-    if (!_textInput) {
+    if (!_resetButton) {
       return;
     }
 
-    _resetButton = document.getElementById('searchForm-reset');
+    _textInput = document.getElementById("searchForm_lookfor");
```

Comment:
`_textInput` should not be initialized here. It's would possible that there is no `_resetButton` but you still need `_textInput` for the virtual keyboard.